proposals/0000-language-editions.rst

```diff
@@ -0,0 +1,1144 @@
+  confusing to contemplate (and I would agree!) but given that in the
+  past we've felt it necessary to change the meaning of extensions
+  from one GHC release to the next it seems plausible that we might
+  want to make changes in the future. We could decide that those kind
+  of changes can only be made by adding a new extension, which would
+  be arguably better.
+
+* Q: What's the binary compatibility guarantee?
+
+  A: The restriction on the expressive power of language editions is that build
+  products of different language editions must be compatible. The compatibility
+  guarantee applies only *within* one release of GHC. That is, GHC 9.12 will
+  produce binaries compatible with other binaries produced by GHC 9.12, regardless
+  of the language edition. However, *no* guarantee is provided between GHC 9.12
+  and GHC 9.14, even under the same language edition.
+
+* Q: With the power embodied in language editions, might we be inviting a
+  state of chaos, as these can evolve arbitrarily over time?
+
+  A: Technically, yes, we're giving ourselves the power to make arbitrary
+  changes from one version of GHC to the next. But this is nothing new! We've
+  always had the ability to change the meaning of language extensions (or the
+  base language) between releases. However, we have striven to make tasteful
+  such changes. We will continue to do so.
+
+
+
+Implementation Plan
+-------------------
+
+This will be easy to implement; I volunteer to do so.
+
+.. [#]
+   With any of the new language editions specified (that is, anything
+   other than ``Haskell98``, ``Haskell2010``, or ``GHC2021``),
+   ``ExistentialQuantification`` will allow only existential quantification,
+   not also (and accidentally) GADTs.
+
+.. [#]
+   ``LexicalNegation`` fixes a wart in the language around unary ``-``, by
+   requiring unary ``-`` not to have any whitespace after it. However, unary
+   ``-`` remains the only symbolic unary function in the language. I hereby
+   propose fixing the wart in the other direction, by just killing off unary
+   ``-``. In the presence of ``NegativeLiterals``, we can still write ``-5``
+   for negative 5. But if you want to negate ``x``, just ``negate x``.
+
+   Thus with any of the new language editions, users cannot write a unary
+   ``-``, unless ``Classic`` mode is on.
+
+.. [#]
+   It is time for the ``Safe`` ecosystem to be dismantled. We should warn
+   on the use of any of the ``Safe`` language extensions, but otherwise
+   ignore them. With the introduction of the language editions proposed
+   here, GHC will allow a ``Safe`` module to depend on any module, thus
+   allowing for a transition period, even if some libraries eagerly adopt
+   these language editions.
+
+ .. [#]
+   These warnings cannot be disabled in ``Stable`` mode, because the mechanism
+   for producing them may conceivably change over time. Recall that packages
+   cannot be released with warnings (without the user disabling this mechanism)
```

Comment:
Is this still part of the proposal, and if so, where?

Reply:
Good point. I've rephrased.